crates/telemetry-subscribers/src/lib.rs

```diff
@@ -469,46 +471,52 @@ impl TelemetryConfig {
                 log: log_filter_handle,
                 trace: trace_filter_handle,
                 file_output,
+                sampler,
             },
         )
     }
 }
 
+// Like Sampler::TraceIdRatioBased, but can be updated at runtime
+#[derive(Debug, Clone)]
 struct SamplingFilter {
-    counter: AtomicUsize,
-    sample_nth: usize,
-    target_prefix: Option<String>,
+    // Sampling filter needs to be fast, so we avoid a mutex.
+    sample_rate: Arc<AtomicF64>,
 }
 
 impl SamplingFilter {
-    fn new(sample_nth: usize, target_prefix: Option<String>) -> Self {
+    fn new(sample_rate: f64) -> Self {
         SamplingFilter {
-            counter: AtomicUsize::new(0),
-            sample_nth,
-            target_prefix,
+            sample_rate: Arc::new(AtomicF64::new(Self::clamp(sample_rate))),
         }
     }
 
-    fn is_sampled(&self) -> bool {
-        self.counter.fetch_add(1, Ordering::Relaxed) % self.sample_nth == 0
+    fn clamp(sample_rate: f64) -> f64 {
+        // clamp sample rate to between 0.0001 and 1.0
+        sample_rate.max(0.0001).min(1.0)
```

Comment:
```suggestion
        sample_rate.max(1.0).min(0.0001)
```
?

Reply:
the original code is correct - min/max is always confusing